src/library/scala/runtime/VM.java

```diff
@@ -0,0 +1,46 @@
+package scala.runtime;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+
+public class VM {
+    public static final MethodHandle RELEASE_FENCE;
+
+
+    static {
+        RELEASE_FENCE = mkHandle();
```

Comment:
This code is protected by the static init lock on this class, so whichever thread gets here first can safely initialize. `RELEASE_FENCE = ...` will be safely published. Do you see this differently?

Reply:
"Permissions" as in a `SecurityManager` sense? I hadn't thought of that. Do you have a recommendation based on your Akka experience?

Reply:
@retronym I don't have any "current" recommendations since I haven't been keeping up with how the module system will impact this situation. Perhaps worth asking on the concurrency-interest ML?

Reply:
If we're running on 9, `lookup.findStatic(Class.forName("java.lang.invoke.VarHandle"), "releaseFence", MethodType.methodType(Void.TYPE));` will work, so we don't need to worry about the module system preventing the fallback case from working.

I guess we could initialize `VM` from `Predef`, but which `Thread` is going to initialize `Predef`? If we do find a problem in the wild, the user will have a workaround of initializing this in their application entry point, and we can revisit.